### ISSUE TITLE
SyntaxError: unreachable code after return statement

### ISSUE THREAD

**Issue by hdon** [moment/moment#2560]
With moment.min.js version 2.10.3 using Firefox 40.0 I get the following message console:

```SyntaxError: unreachable code after return statement```

**Comment by cvcv01** [comment]
I get that too

**Comment by aesedepece** [comment]
As read in MDN:

> **Automatic Semicolon Insertion**
> The return statement is affected by automatic semicolon insertion (ASI). No line terminator is allowed between the return keyword and the expression.
>```javascript
>return
>a + b;
>```
>is transformed by ASI into:
>```javascript
>return; 
>a + b;
>```
>Starting with Gecko 40 the console will warn "unreachable code after return statement".

**Comment by zoechi** [comment]
Looks like https://bugzilla.mozilla.org/show_bug.cgi?id=1151931

**Comment by mj1856** [comment]
Reproducable with current moment 2.10.6.

**Comment by mj1856** [comment]
Actually, I take that back.  That was coming from jsFiddle's own code.  I don't see this problem with moment by itself.

Is there a specific function that gives this error?

**Comment by hdon** [comment]
>Is there a specific function that gives this error?

I don't have access to the bug right this second, but it happens when the script is loaded, not when any function is called.

**Comment by mj1856** [comment]
@hdon  - Would be helpful to know if it still errors for you in the current version (2.10.6).  Thanks.

**Comment by olivierMonvoisin** [comment]
I have the same issue. Because of that, in the dev tool, the breakpoint doesnt hint.

**Comment by adamelso** [comment]
I have the same issue with Firefox 40.0.3. In Chrome, the error doesn't exist.

**Comment by ichernev** [comment]
Can somebody provide easy-to-reproduce steps? If its only the minified version, its a problem with the minifier/firefox, rather than moment. But minified won't have any new lines (and if it does I hope they're not after a `return`).

**Comment by ichernev** [comment]
Do the tests pass? If they do than it can not be a `return\na + b;`, because this changes semantics.

Note: tests on momentjs.com use non-minified versions, somebody try them out with minified.

**Comment by ichernev** [comment]
The code was wrong from the rewrite `2.10.0` to version `2.10.3`, function `cloneWithOffset` had 2 returns in a row.

This issue was fixed in `2.10.5` (there was no `2.10.4`).

**Comment by sjlongland** [comment]
Hate to bump an old issue, but I'm hitting this very same issue with `moment` 2.24.0:

```
unreachable code after return statement[Learn More] build.js:5861:8
unreachable code after return statement[Learn More] build.js line 870 > eval:5861:8
unreachable code after return statement[Learn More] build.js:5848:8
unreachable code after return statement[Learn More] build.js line 4377 > eval:5848:8
```

When I go digging through the whopping monster that is `build.js`, I see:

```
/* line 5854-5861 */
/***/ "./node_modules/moment/locale/es-do.js":
/*!*********************************************!*\
  !*** ./node_modules/moment/locale/es-do.js ***!
  \*********************************************/
/*! no static exports found */
/***/ (function(module, exports, __webpack_require__) {

eval("var __WEBPACK_AMD_DEFINE_FACTORY__, __WEBPACK_AMD_DEFINE_ARRAY__, __WEBPACK_AMD_DEFINE_RESULT__;function _typeof(obj) { if (typeof Symbol === \"function\" && typeof Symbol.iterator === \"symbol\") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === \"function\" && obj.constructor === Symbol && obj !== Symbol.prototype ? \"symbol\" : typeof obj; }; } return _typeof(obj); }\n\n//! moment.js locale configuration\n;\n\n(function (global, factory) {\n  ( false ? undefined : _typeof(exports)) === 'object' && typeof module !== 'undefined' && \"function\" === 'function' ? factory(__webpack_require__(/*! ../moment */ \"./node_modules/moment/moment.js\")) :  true ? !(__WEBPACK_AMD_DEFINE_ARRAY__ = [__webpack_require__(/*! ../moment */ \"./node_modules/moment/moment.js\")], __WEBPACK_AMD_DEFINE_FACTORY__ = (factory),\n\t\t\t\t__WEBPACK_AMD_DEFINE_RESULT__ = (typeof __WEBPACK_AMD_DEFINE_FACTORY__ === 'function' ?\n\t\t\t\t(__WEBPACK_AMD_DEFINE_FACTORY__.apply(exports, __WEBPACK_AMD_DEFINE_ARRAY__)) : __WEBPACK_AMD_DEFINE_FACTORY__),\n\t\t\t\t__WEBPACK_AMD_DEFINE_RESULT__ !== undefined && (module.exports = __WEBPACK_AMD_DEFINE_RESULT__)) : undefined;\n})(this, function (moment) {\n  'use strict';\n\n  var monthsShortDot = 'ene._feb._mar._abr._may._jun._jul._ago._sep._oct._nov._dic.'.split('_'),\n      _monthsShort = 'ene_feb_mar_abr_may_jun_jul_ago_sep_oct_nov_dic'.split('_');\n\n  var monthsParse = [/^ene/i, /^feb/i, /^mar/i, /^abr/i, /^may/i, /^jun/i, /^jul/i, /^ago/i, /^sep/i, /^oct/i, /^nov/i, /^dic/i];\n  var monthsRegex = /^(enero|febrero|marzo|abril|mayo|junio|julio|agosto|septiembre|octubre|noviembre|diciembre|ene\\.?|feb\\.?|mar\\.?|abr\\.?|may\\.?|jun\\.?|jul\\.?|ago\\.?|sep\\.?|oct\\.?|nov\\.?|dic\\.?)/i;\n  var esDo = moment.defineLocale('es-do', {\n    months: 'enero_febrero_marzo_abril_mayo_junio_julio_agosto_septiembre_octubre_noviembre_diciembre'.split('_'),\n    monthsShort: function monthsShort(m, format) {\n      if (!m) {\n        return monthsShortDot;\n      } else if (/-MMM-/.test(format)) {\n        return _monthsShort[m.month()];\n      } else {\n        return monthsShortDot[m.month()];\n      }\n    },\n    monthsRegex: monthsRegex,\n    monthsShortRegex: monthsRegex,\n    monthsStrictRegex: /^(enero|febrero|marzo|abril|mayo|junio|julio|agosto|septiembre|octubre|noviembre|diciembre)/i,\n    monthsShortStrictRegex: /^(ene\\.?|feb\\.?|mar\\.?|abr\\.?|may\\.?|jun\\.?|jul\\.?|ago\\.?|sep\\.?|oct\\.?|nov\\.?|dic\\.?)/i,\n    monthsParse: monthsParse,\n    longMonthsParse: monthsParse,\n    shortMonthsParse: monthsParse,\n    weekdays: 'domingo_lunes_martes_miércoles_jueves_viernes_sábado'.split('_'),\n    weekdaysShort: 'dom._lun._mar._mié._jue._vie._sáb.'.split('_'),\n    weekdaysMin: 'do_lu_ma_mi_ju_vi_sá'.split('_'),\n    weekdaysParseExact: true,\n    longDateFormat: {\n      LT: 'h:mm A',\n      LTS: 'h:mm:ss A',\n      L: 'DD/MM/YYYY',\n      LL: 'D [de] MMMM [de] YYYY',\n      LLL: 'D [de] MMMM [de] YYYY h:mm A',\n      LLLL: 'dddd, D [de] MMMM [de] YYYY h:mm A'\n    },\n    calendar: {\n      sameDay: function sameDay() {\n        return '[hoy a la' + (this.hours() !== 1 ? 's' : '') + '] LT';\n      },\n      nextDay: function nextDay() {\n        return '[mañana a la' + (this.hours() !== 1 ? 's' : '') + '] LT';\n      },\n      nextWeek: function nextWeek() {\n        return 'dddd [a la' + (this.hours() !== 1 ? 's' : '') + '] LT';\n      },\n      lastDay: function lastDay() {\n        return '[ayer a la' + (this.hours() !== 1 ? 's' : '') + '] LT';\n      },\n      lastWeek: function lastWeek() {\n        return '[el] dddd [pasado a la' + (this.hours() !== 1 ? 's' : '') + '] LT';\n      },\n      sameElse: 'L'\n    },\n    relativeTime: {\n      future: 'en %s',\n      past: 'hace %s',\n      s: 'unos segundos',\n      ss: '%d segundos',\n      m: 'un minuto',\n      mm: '%d minutos',\n      h: 'una hora',\n      hh: '%d horas',\n      d: 'un día',\n      dd: '%d días',\n      M: 'un mes',\n      MM: '%d meses',\n      y: 'un año',\n      yy: '%d años'\n    },\n    dayOfMonthOrdinalParse: /\\d{1,2}º/,\n    ordinal: '%dº',\n    week: {\n      dow: 1,\n      // Monday is the first day of the week.\n      doy: 4 // The week that contains Jan 4th is the first week of the year.\n\n    }\n  });\n  return esDo;\n});\n\n//# sourceURL=webpack:///./node_modules/moment/locale/es-do.js?");

/* 5843-5848 */
```
/***/ "./node_modules/moment/locale/eo.js":
/*!******************************************!*\
  !*** ./node_modules/moment/locale/eo.js ***!
  \******************************************/
/*! no static exports found */
/***/ (function(module, exports, __webpack_require__) {
```

Something about that code, once Webpack stuffs it into a single `.js` file, upsets Firefox.  I haven't figured out a way to silence the warning because _this isn't my code_.
```
